components/AvalancheCenterSelector.tsx

```diff
@@ -43,56 +44,88 @@ const center_idsByType: SectionListData<string>[] = [
 ];
 
 interface AvalancheCenterCardProps {
-  center_id: string;
-  set_avalanche_center: React.Dispatch<React.SetStateAction<string>>;
+  avalancheCenterId: string;
+  selected: boolean;
+  onPress: (center_id: string) => void;
 }
 
-const AvalancheCenterCard: React.FunctionComponent<AvalancheCenterCardProps> = ({center_id, set_avalanche_center}: AvalancheCenterCardProps) => {
-  const navigation = useNavigation<TabNavigationProps>();
-  const {isLoading, isError, data: avalancheCenter, error} = useAvalancheCenterMetadata(center_id);
+export const AvalancheCenterCard: React.FunctionComponent<AvalancheCenterCardProps> = ({avalancheCenterId, selected, onPress}: AvalancheCenterCardProps) => {
+  const {width, height} = useWindowDimensions();
+  const {isLoading, isError, data: avalancheCenter, error} = useAvalancheCenterMetadata(avalancheCenterId);
   if (isLoading) {
     return <ActivityIndicator style={styles.item} />;
   }
   if (isError) {
     return (
-      <View style={styles.item}>
-        <Text>{`Could not fetch data for ${center_id}: ${error?.message}.`}</Text>
-      </View>
+      <Box>
+        <Box bg="light.100">
+          <VStack>
+            <HStack justifyContent="flex-start" alignItems="center">
+              <NBText color="light.400">{`Could not fetch data for ${avalancheCenterId}: ${error?.message}.`}</NBText>
+            </HStack>
+          </VStack>
+        </Box>
+      </Box>
     );
   }
   if (!avalancheCenter) {
     return (
-      <View style={styles.item}>
-        <Text>{`No metadata found for ${center_id}.`}</Text>
-      </View>
+      <Box>
+        <Box bg="light.100">
+          <VStack>
+            <HStack justifyContent="flex-start" alignItems="center">
+              <NBText color="light.400">{`No metadata found for ${avalancheCenterId}.`}</NBText>
+            </HStack>
+          </VStack>
+        </Box>
+      </Box>
     );
   }
 
   return (
-    <TouchableOpacity
-      style={styles.item}
-      onPress={() => {
-        set_avalanche_center(center_id);
-        // We need to clear navigation state to force all screens from the
-        // previous avalanche center selection to unmount
-        navigation.reset({
-          index: 0,
-          routes: [{name: 'Home'}],
-        });
-      }}>
-      <AvalancheCenterLogo style={styles.logo} center_id={center_id} />
-      <Text style={{textAlignVertical: 'center'}}>{avalancheCenter.name}</Text>
-    </TouchableOpacity>
+    <Box>
+      <TouchableOpacity
+        style={styles.item}
+        onPress={() => {
+          onPress(avalancheCenterId);
+        }}>
+        <Box bg={selected ? 'blue.100' : 'light.100'}>
+          <VStack>
+            <HStack justifyContent="flex-start" alignItems="center" px="2" width={width}>
+              <AvalancheCenterLogo style={{height: height / 20}} center_id={avalancheCenterId} />
```

Comment:
can't you just do `height: '5%'`? (I'm also kind of skeptical that we want the height of any icons to change based on screen size, they should be coupled to the size of the text that they're next to)

Reply:
Love the idea - can you show me how to make it relative to the font?